lib/src/sketch_schemes/mash.rs

```diff
@@ -67,10 +67,10 @@ impl MashSketcher {
 
 impl SketchScheme for MashSketcher {
     fn process(&mut self, seq: &SequenceRecord) {
-        self.total_bases += seq.sequence().len() as u64;
-        let rc = seq.reverse_complement();
-        for (_, kmer, is_rev_complement) in
-            seq.normalize(false).canonical_kmers(self.kmer_length, &rc)
+        let norm_seq = seq.normalize(false);
```

Comment:
I think in practice this trait is not well thought. The `process` function should take a `&Sequence` since we do not need anything from the `&SequenceRecord` and normalizing it here implicitly would allocate twice for normalized sequence if you need that elsewhere. In that case you would still need to call `normalize` before calling `process` though.